projects/ion/src/lib/message/message.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, HostBinding, Input, OnInit } from '@angular/core';
+import { IonMessageProps } from './types';
+import { IonIconComponent } from '../icon';
+
+const icontypes = {
```

Comment:
If you type this correctly, you won't need to do the type assertion on line 29.

```suggestion
const icontypes: Record<IonMessageProps['type'], IonMessageProps['iconType']> = {
```